onnxruntime/python/tools/quantization/calibrate_user.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+# coding: utf-8
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft, Intel Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import os
+import sys
+import numpy as np
+import re
+import subprocess
+import json
+
+import abc
+
+import onnx
+import onnxruntime
+from onnx import helper, TensorProto, numpy_helper
+from quantize import quantize, QuantizationMode
+from calibrate import calibrate
+
+#user-implement preprocess func
+from data_preprocess import preprocess_func
+
+
+class CalibrationDataReaderInterface(metaclass=abc.ABCMeta):
+    @classmethod
+    def __subclasshook__(cls,subclass):
+        return (hasattr(subclass,'get_next') and callable(subclass.get_next) or NotImplemented)
+
+    @abc.abstractmethod
+    def get_next(self) -> dict:
+        """generate the input data dict for ONNXinferenceSession run"""
+        raise NotImplementedError
+
+class CalibrationDataReader(CalibrationDataReaderInterface):
```

Comment:
>CalibrationDataReader [](start = 6, length = 21)

after changing CalibrationDataReaderInterface to CalibrationDataReader, you can name this one with specific name, like ResNet50DataReader